### ISSUE TITLE
[33] Learning to Prompt for Continual Learning

### ISSUE THREAD

**Issue by long8v** [long8v/PTIR#38]
<img width="1407" alt="image" src="https://user-images.githubusercontent.com/46675408/178699787-23cc2115-59a8-47eb-953b-19ae51eaeffb.png">

[paper](https://arxiv.org/pdf/2112.08654.pdf)

## TL;DR
- **task :** class incremental learning / domain incremental learning / task-agnostic learning
- **problem :** catastrophic forgetting. 
- **idea :** prompt learning ! 이미지가 주어졌을 때 전체 M개의 prompt pool 중에 N개의 가까운 prompt를 뽑고(Learning to Prompt, L2P), ViT 비전 토큰 앞에 prepend해서 이미지 분류. 
- **architecture :** ViT-B/16
- **objective :** CrossEntropyLoss + diversifying prompt-selection.  
- **baseline :** CIL variants(finetuning sequentially, BiC, EWC, DER++..)
- **data :** split CIFAR100, 5-datasets
- **result :** SOTA. 그냥 iid finetuning보단 낮음. 
- **contribution :** 간단한 아이디어/아키텍쳐로 SOTA. 

### Preliminaries 
- Taxonomy of CIL

- idea
<img width="1069" alt="image" src="https://user-images.githubusercontent.com/46675408/178700683-6af3e862-b05c-40ce-a6ab-54f1489aa876.png">

- architecture
ViT

거리는 cosine 으로 했더니 잘됐다. 

이때 token은 key-value 구조.

loss 
opto
 balancing 하기 위한 loss도 있음
